tethys_apps/utilities.py

```diff
@@ -701,3 +707,35 @@ def sign_and_unsign_secret_string(signer, value, is_signing):
     else:
         secret_unsigned = signer.unsign_object(f"{value}")
         return secret_unsigned
+
+
+def update_decorated_websocket_consumer_class(
+    function_or_class, permissions_required, permissions_use_or, login_required
+):
+    """Updates a given consumer class and adds the necessary properties and function for authorizing user access
+    depending on the other args given.
+
+    Args:
+        function_or_class (class): class of the websocket consumer
+        permissions_required (str, list, tuple): the permissions required for user access
+        permissions_use_or (bool): Determines if all permissions need to be met or just one of them
+        login_required (bool): Determines if the user needs to be logged in to use
+
+    Returns:
+        class: updated class with necessary properties and function for authorizing user access
+    """
+    base_class_name = inspect.getmro(function_or_class)[1].__name__.lower()
+    function_or_class_name = function_or_class.__name__.lower()
+    if "async" in function_or_class_name or "async" in base_class_name:
```

Comment:
I suppose this approach isn't foolproof either. Someone could do something like:

```python
from channels.consumer import SyncConsumer, AsyncConsumer

class AnAsyncConsumer(AsyncConsumer):
   pass

class ASyncConsumer(SyncConsumer):
   pass

```
 
You never know! 🤷 

According to the Tethys Docs:

> The consumer decorator functions largely the same way as the controller decorator except that it is used to decorate a consumer class, which must be a subclass of either channels.consumer.AsyncConsumer or channels.consumer.SyncConsumer

I'm hoping that I was smarter when I wrote that than I am now. If it is indeed true that a consumer class must be a subclass of  `channels.consumer.AsyncConsumer` then:

```python
if issubclass(function_or_class, channels.consumer.AsyncConsumer):
    consumer_mixin = TethysAsyncWebsocketConsumerMixin
```

Reply:
Thanks Scott. Those are valid points. I like the final proposed solution. Simple and clean for sure.

Reply:
Although, maybe that won't work. I just looked at my implementation and I can using the AsyncWebsocketConsumer class from from channels.generic.websocket. That is also what is suggested in the tutorials for the websockets.

Reply:
The `AsyncWebsocketConsumer` is a subclass of `AsyncConsumer`. I think the `issubclass` will look at the whole parent chain, but I'm not sure.

Reply:
```python
from channels.generic.websocket import AsyncWebsocketConsumer
class MyConsumer(AsyncWebsocketConsumer):
    groups = ["broadcast"]
    
from channels.consumer import AsyncConsumer
issubclass(MyConsumer, AsyncConsumer)
True
```

Reply:
just implemented this. I had to do it the opposite way and checked if it is a subclass of `SyncConsumer` because the `WebsocketConsumer` and `AsyncWebsocketConsumer` are both subclasses of `AsyncConsumer` apparently.